pkg/authorization/oauth/context.go

```diff
@@ -16,6 +16,22 @@ func (c *IntrospectionContext) IsAuthorized() bool {
 	return c.IntrospectionResponse.Active
 }
 
+// OrganizationID implements [authorization.Ctx] by returning the `urn:zitadel:iam:user:resourceowner:id` claim or the `urn:zitadel:iam:org:id:{id}` claim of the [oidc.IntrospectionResponse].
+func (c *IntrospectionContext) OrganizationID() string {
+	if c == nil {
+		return ""
+	}
+	// check for organization ID when using scope "urn:zitadel:iam:user:resourceowner"
+	if orgID, ok := c.IntrospectionResponse.Claims["urn:zitadel:iam:user:resourceowner:id"].(string); ok {
+		return orgID
+	}
+	// check for organization ID when using scope "urn:zitadel:iam:org:id:{id}"
+	if orgID, ok := c.IntrospectionResponse.Claims["urn:zitadel:iam:org:id"].(string); ok {
+		return orgID
+	}
+	return ""
```

Comment:
```suggestion
// OrganizationID implements [authorization.Ctx] by returning the `urn:zitadel:iam:user:resourceowner:id` claim of the [oidc.IntrospectionResponse].
func (c *IntrospectionContext) OrganizationID() string {
	if c == nil {
		return ""
	}
	// check for organization ID when using scope "urn:zitadel:iam:user:resourceowner"
	orgID, _ := c.IntrospectionResponse.Claims["urn:zitadel:iam:user:resourceowner:id"].(string)
	return orgID
```
The urn:zitadel:iam:org:id:{id} will enforce the user to be part of the specified organization, but not returned as a claim. The organizationID will also be returned as the urn:zitadel:iam:user:resourceowner:id claim.